actionpack/lib/action_dispatch/middleware/server_timing.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require "action_dispatch/middleware/server_timing_subscriber"
+
+module ActionDispatch
+  class ServerTiming
+    class << self
+      def add_timing(key:, dur: nil, desc: nil)
+        timings[key] = { dur: dur, desc: desc }
+      end
+
+      def to_header
+        headers = []
+        timings.each_pair do |key, data|
+          header = "#{key}"
+          header += ";desc=\"#{data[:desc]}\"" if data[:desc]
+          header += ";dur=#{data[:dur]}" if data[:dur]
```

Comment:
I think `<<` on strings is faster, right?

Reply:
likely, but then we can't use the `# frozen_string_literal: true` magic comment.

Reply:
Under the influence of `# frozen_string_literal: true`, we usually use `header = +key.to_s` for making mutable string.

Reply:
TIL, thanks!